astropy/table/operations.py

```diff
@@ -38,18 +36,21 @@ def _get_list_of_tables(tables):
     Check that tables is a Table or sequence of Tables.  Returns the
     corresponding list of Tables.
     """
-    from .table import Table, Row
 
     # Make sure we have a list of things
     if not isinstance(tables, collections.Sequence):
         tables = [tables]
 
-    # Make sure each thing is a Table or Row
-    if any(not isinstance(x, (Table, Row)) for x in tables) or len(tables) == 0:
-        raise TypeError('`tables` arg must be a Table or sequence of Tables or Rows')
+    # Make sure each thing is a Table or Row or Column
+    if any(not (isinstance(x, (Table, Row, Column)) or has_info_class(x, MixinInfo))
+           for x in tables) or len(tables) == 0:
+        raise TypeError('`tables` arg must be a Table or sequence of Tables or Rows or Columns')
 
-    # Convert any Rows to Tables
-    tables = [(x if isinstance(x, Table) else Table(x)) for x in tables]
+    for ii, tbl in enumerate(tables):
+        if isinstance(tbl, Row):
+            tables[ii] = Table(tbl)
+        elif isinstance(tbl, Column) or has_info_class(tbl, MixinInfo):
+            tables[ii] = Table([tbl])
```

Comment:
One annoyance (special case) that should be handled is if the mixin is a Quantity, then you want a QTable here.  

This should really be handled instead via #1783 by having the stacking be class methods.  Then there is no guessing involved since the user provides the output class.  But for now if they stack a Quantity then the output should be a QTable.

Reply:
and only for Quantities, or also for Time and SkyCoord? I have to admit that still trying to get my head around some of the internals (well, you've seen the evidence for this already).

Reply:
Looking at this code I think it just needs a refactor and can be more general to accept anything column-like.  So e.g. you could do `hstack([[1, 2], ['a', b'], t])`.  Here is some untested code:
```
    # Make sure each thing is a Table or Row or Column
    if len(tables) == 0:
        raise ValueError('no values provided to stack')

    # Convert inputs (Table, Row, or anything column-like) to Tables.
    # Special case that Quantity converts to a QTable.
    for ii, val in enumerate(tables):
        if isinstance(val, Table):
            pass
        elif isinstance(val, Row):
            tables[ii] = Table(val)
        elif isinstance(val, u.Quantity):
            tables[ii] = QTable([val])
        else:
            try:
                tables[ii] = Table([val])
            except Exception:
                raise TypeError('cannot convert {} to table column'
                                .format(val))
```